mod.d/enadiag.yaml

```diff
@@ -0,0 +1,270 @@
+      Return a list of non-loopback interfaces on the system.
+
+      Returns:
+          interface_list (list): List of interfaces (e.g. ["eth0", "eth1"]
+      """
+      try:
+          interface_list = []
+          # Get all net interfaces except loopback, append to list, sort it.
+          for interface in os.listdir("/sys/class/net"):
+              if interface != "lo":
+                  interface_list.append(interface)
+          if len(interface_list) > 0:
+              interface_list = sorted(interface_list)
+          # Return an error if the list is somehow empty (didn't find any network interfaces)
+          else:
+              raise ValueError("No interfaces found.")
+      # Catch OSError in Python2 and FileNotFoundError in Python3 (inherits from OSError)
+      except OSError:
+          print("Unable to build interface list.")
+      return interface_list
+
+  # Create list with get_interfaces and create Interface class objects with them.
+  interfaces = get_interfaces()
+  interface_results = [Interface(name) for name in interfaces]
+
+  def get_ena_version():
+      """
+      Return the version of the driver for the first alphabetically ordered non-virtual network interface.
+
+      Returns:
+          driver_version (str): version os driver (e.g. 2.2.4g)
+      """
+      try:
+          interface_list = []
+          # Add all non-virtual interfaces to the list and sort it
+          for interface in os.listdir("/sys/class/net"):
+              if "virtual" not in os.path.abspath(os.readlink("/sys/class/net/{}".format(interface))):
+                  interface_list.append(interface)
+          if len(interface_list) > 0:
+              interface_list = sorted(interface_list)
+              # Open the file, strip out newline characters
+              with open ("/sys/class/net/{}/device/driver/module/version".format(interface_list[0]), "r") as file:
+                  driver_version = file.read().replace('\n', '')
+          # Return an error if the list is somehow empty (didn't find any network interfaces)
+          else:
+              driver_version = "Unknown"
+      # Catch OSError in Python2 and FileNotFoundError in Python3 (inherits from OSError)
+      except OSError:
+          driver_version = "Unknown"
+      return driver_version
+
+  def compare_ena_version():
+      """
+      Check to see if the ena version is new enough to have the necessary stats available.
+
+      Returns:
+          Boolean True if version is high enough
+      """
+      # Define version with stats included
+      stats_included = "2.2.10"
+      current_version = get_ena_version()
+      # Get only the version number and remove trailing letters to make version numbers comparable
+      comprable_version = re.search("(\d+.\d+.\d+)", current_version).group(0)
+      return LooseVersion(comprable_version) > LooseVersion(stats_included)
+
+  def get_ena_stats(interface):
+      """
+      Uses subprocess to pull ethtool statisics
```

Comment:
Please document the version(s) of ethtool tested.

Reply:
Will do